pennylane_lightning/src/Gates.hpp

```diff
@@ -323,6 +323,23 @@ namespace Pennylane {
         void applyKernel(const StateVector& state, const std::vector<size_t>& indices, const std::vector<size_t>& externalIndices);
     };
 
+    class Unitary : public AbstractGate {
+    private:
+        const std::vector<CplxType> matrix;
+        const std::vector<double> real_matrix;
```

Comment:
Worth leaving a comment on how this vector represents the complex matrix for reference.